### ISSUE TITLE
[6.0.0-prerelease.1] WCT fails to launch Safari 10.0.1 when testing variants

### ISSUE THREAD

**Issue by cdata** [Polymer/web-component-tester#459]
### Description

When running WCT ~5, Safari 10.0.1 can be successfully launched.
When running WCT 6.0.0-prerelease.1 to test a repo without variants, Safari 10 can be successfully launched.
When running WCT 6.0.0-prerelease.1 to test a repo with variants, Safari 10 cannot be launched.

### Steps to reproduce

 1. Setup a repo for a Polymer Elements project that supports variants. Example: 
```
git clone git@github.com:polymerelements/paper-button.git
cd ./paper-button
git checkout --track origin/2.0-preview
```
 2. Install the variants: `polymer install --variants`
 3. Run wct: `wct`

### Expected outcome

WCT launches Safari 10 and runs the tests.

### Actual outcome

WCT shows an error and Safari 10 is never launched. Other browsers are (most likely) launched successfully.

### Example error

```
Error: {"state":"unknown error","sessionId":null,"hCode":784386304,"value":{"localizedMessage":"Connection refused","cause":null,"stackTrace":[{"fileName":"PlainSocketImpl.java","nativeMethod":true,"methodName":"socketConnect","className":"java.net.PlainSocketImpl","hCode":1566792002,"lineNumber":-2,"class":"java.lang.StackTraceElement"},{"fileName":"AbstractPlainSocketImpl.java","nativeMethod":false,"methodName":"doConnect","className":"java.net.AbstractPlainSocketImpl","hCode":495250506,"lineNumber":350,"class":"java.lang.StackTraceElement"},{"fileName":"AbstractPlainSocketImpl.java","nativeMethod":false,"methodName":"connectToAddress","className":"java.net.AbstractPlainSocketImpl","hCode":-2104362742,"lineNumber":206,"class":"java.lang.StackTraceElement"},{"fileName":"AbstractPlainSocketImpl.java","nativeMethod":false,"methodName":"connect","className":"java.net.AbstractPlainSocketImpl","hCode":-1864045581,"lineNumber":188,"class":"java.lang.StackTraceElement"},{"fileName":"SocksSocketImpl.java","nativeMethod":false,"methodName":"connect","className":"java.net.SocksSocketImpl","hCode":1780853535,"lineNumber":392,"class":"java.lang.StackTraceElement"},{"fileName":"Socket.java","nativeMethod":false,"methodName":"connect","className":"java.net.Socket","hCode":1700005482,"lineNumber":589,"class":"java.lang.StackTraceElement"},{"fileName":"PlainConnectionSocketFactory.java","nativeMethod":false,"methodName":"connectSocket","className":"org.apache.http.conn.socket.PlainConnectionSocketFactory","hCode":1175278798,"lineNumber":74,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultHttpClientConnectionOperator.java","nativeMethod":false,"methodName":"connect","className":"org.apache.http.impl.conn.DefaultHttpClientConnectionOperator","hCode":-502593941,"lineNumber":141,"class":"java.lang.StackTraceElement"},{"fileName":"PoolingHttpClientConnectionManager.java","nativeMethod":false,"methodName":"connect","className":"org.apache.http.impl.conn.PoolingHttpClientConnectionManager","hCode":-1370960713,"lineNumber":353,"class":"java.lang.StackTraceElement"},{"fileName":"MainClientExec.java","nativeMethod":false,"methodName":"establishRoute","className":"org.apache.http.impl.execchain.MainClientExec","hCode":848422720,"lineNumber":380,"class":"java.lang.StackTraceElement"},{"fileName":"MainClientExec.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.execchain.MainClientExec","hCode":1688513621,"lineNumber":236,"class":"java.lang.StackTraceElement"},{"fileName":"ProtocolExec.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.execchain.ProtocolExec","hCode":-642149983,"lineNumber":184,"class":"java.lang.StackTraceElement"},{"fileName":"RetryExec.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.execchain.RetryExec","hCode":2007799545,"lineNumber":88,"class":"java.lang.StackTraceElement"},{"fileName":"RedirectExec.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.execchain.RedirectExec","hCode":608278487,"lineNumber":110,"class":"java.lang.StackTraceElement"},{"fileName":"InternalHttpClient.java","nativeMethod":false,"methodName":"doExecute","className":"org.apache.http.impl.client.InternalHttpClient","hCode":2023045549,"lineNumber":184,"class":"java.lang.StackTraceElement"},{"fileName":"CloseableHttpClient.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.client.CloseableHttpClient","hCode":1628029009,"lineNumber":71,"class":"java.lang.StackTraceElement"},{"fileName":"CloseableHttpClient.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.client.CloseableHttpClient","hCode":1628028993,"lineNumber":55,"class":"java.lang.StackTraceElement"},{"fileName":"ApacheHttpClient.java","nativeMethod":false,"methodName":"fallBackExecute","className":"org.openqa.selenium.remote.internal.ApacheHttpClient","hCode":652052258,"lineNumber":142,"class":"java.lang.StackTraceElement"},{"fileName":"ApacheHttpClient.java","nativeMethod":false,"methodName":"execute","className":"org.openqa.selenium.remote.internal.ApacheHttpClient","hCode":-904183858,"lineNumber":88,"class":"java.lang.StackTraceElement"},{"fileName":"ProtocolHandshake.java","nativeMethod":false,"methodName":"createSession","className":"org.openqa.selenium.remote.ProtocolHandshake","hCode":-960127848,"lineNumber":108,"class":"java.lang.StackTraceElement"},{"fileName":"ProtocolHandshake.java","nativeMethod":false,"methodName":"createSession","className":"org.openqa.selenium.remote.ProtocolHandshake","hCode":-960127892,"lineNumber":64,"class":"java.lang.StackTraceElement"},{"fileName":"HttpCommandExecutor.java","nativeMethod":false,"methodName":"execute","className":"org.openqa.selenium.remote.HttpCommandExecutor","hCode":-1852790188,"lineNumber":141,"class":"java.lang.StackTraceElement"},{"fileName":"DriverCommandExecutor.java","nativeMethod":false,"methodName":"execute","className":"org.openqa.selenium.remote.service.DriverCommandExecutor","hCode":-633333582,"lineNumber":82,"class":"java.lang.StackTraceElement"},{"fileName":"RemoteWebDriver.java","nativeMethod":false,"methodName":"execute","className":"org.openqa.selenium.remote.RemoteWebDriver","hCode":1767885088,"lineNumber":601,"class":"java.lang.StackTraceElement"},{"fileName":"RemoteWebDriver.java","nativeMethod":false,"methodName":"startSession","className":"org.openqa.selenium.remote.RemoteWebDriver","hCode":-1182648233,"lineNumber":241,"class":"java.lang.StackTraceElement"},{"fileName":"RemoteWebDriver.java","nativeMethod":false,"methodName":"<init>","className":"org.openqa.selenium.remote.RemoteWebDriver","hCode":2001602308,"lineNumber":128,"class":"java.lang.StackTraceElement"},{"fileName":"SafariDriver.java","nativeMethod":false,"methodName":"<init>","className":"org.openqa.selenium.safari.SafariDriver","hCode":-1792744904,"lineNumber":62,"class":"java.lang.StackTraceElement"},{"fileName":"SafariDriver.java","nativeMethod":false,"methodName":"<init>","className":"org.openqa.selenium.safari.SafariDriver","hCode":-1792744913,"lineNumber":53,"class":"java.lang.StackTraceElement"},{"fileName":"NativeConstructorAccessorImpl.java","nativeMethod":true,"methodName":"newInstance0","className":"sun.reflect.NativeConstructorAccessorImpl","hCode":-917798116,"lineNumber":-2,"class":"java.lang.StackTraceElement"},{"fileName":"NativeConstructorAccessorImpl.java","nativeMethod":false,"methodName":"newInstance","className":"sun.reflect.NativeConstructorAccessorImpl","hCode":-1864225098,"lineNumber":62,"class":"java.lang.StackTraceElement"},{"fileName":"DelegatingConstructorAccessorImpl.java","nativeMethod":false,"methodName":"newInstance","className":"sun.reflect.DelegatingConstructorAccessorImpl","hCode":-2122307259,"lineNumber":45,"class":"java.lang.StackTraceElement"},{"fileName":"Constructor.java","nativeMethod":false,"methodName":"newInstance","className":"java.lang.reflect.Constructor","hCode":-1319859919,"lineNumber":423,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultDriverProvider.java","nativeMethod":false,"methodName":"callConstructor","className":"org.openqa.selenium.remote.server.DefaultDriverProvider","hCode":-2096353668,"lineNumber":103,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultDriverProvider.java","nativeMethod":false,"methodName":"newInstance","className":"org.openqa.selenium.remote.server.DefaultDriverProvider","hCode":1201680335,"lineNumber":97,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultDriverFactory.java","nativeMethod":false,"methodName":"newInstance","className":"org.openqa.selenium.remote.server.DefaultDriverFactory","hCode":-1365021416,"lineNumber":60,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultSession.java","nativeMethod":false,"methodName":"call","className":"org.openqa.selenium.remote.server.DefaultSession$BrowserCreator","hCode":2056168471,"lineNumber":222,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultSession.java","nativeMethod":false,"methodName":"call","className":"org.openqa.selenium.remote.server.DefaultSession$BrowserCreator","hCode":2056168458,"lineNumber":209,"class":"java.lang.StackTraceElement"},{"fileName":"FutureTask.java","nativeMethod":false,"methodName":"run","className":"java.util.concurrent.FutureTask","hCode":-820313554,"lineNumber":266,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultSession.java","nativeMethod":false,"methodName":"run","className":"org.openqa.selenium.remote.server.DefaultSession$1","hCode":-255143467,"lineNumber":176,"class":"java.lang.StackTraceElement"},{"fileName":"ThreadPoolExecutor.java","nativeMethod":false,"methodName":"runWorker","className":"java.util.concurrent.ThreadPoolExecutor","hCode":-1285150184,"lineNumber":1142,"class":"java.lang.StackTraceElement"},{"fileName":"ThreadPoolExecutor.java","nativeMethod":false,"methodName":"run","className":"java.util.concurrent.ThreadPoolExecutor$Worker","hCode":-242190645,"lineNumber":617,"class":"java.lang.StackTraceElement"},{"fileName":"Thread.java","nativeMethod":false,"methodName":"run","className":"java.lang.Thread","hCode":-82946320,"lineNumber":745,"class":"java.lang.StackTraceElement"}],"suppressed":[],"message":"Connection refused","hCode":531229532,"class":"java.net.ConnectException","screen":null},"class":"org.openqa.selenium.remote.Response","status":13}, {"state":"unknown error","sessionId":null,"hCode":1366840724,"value":{"localizedMessage":"Connection refused","cause":null,"stackTrace":[{"fileName":"PlainSocketImpl.java","nativeMethod":true,"methodName":"socketConnect","className":"java.net.PlainSocketImpl","hCode":1566792002,"lineNumber":-2,"class":"java.lang.StackTraceElement"},{"fileName":"AbstractPlainSocketImpl.java","nativeMethod":false,"methodName":"doConnect","className":"java.net.AbstractPlainSocketImpl","hCode":495250506,"lineNumber":350,"class":"java.lang.StackTraceElement"},{"fileName":"AbstractPlainSocketImpl.java","nativeMethod":false,"methodName":"connectToAddress","className":"java.net.AbstractPlainSocketImpl","hCode":-2104362742,"lineNumber":206,"class":"java.lang.StackTraceElement"},{"fileName":"AbstractPlainSocketImpl.java","nativeMethod":false,"methodName":"connect","className":"java.net.AbstractPlainSocketImpl","hCode":-1864045581,"lineNumber":188,"class":"java.lang.StackTraceElement"},{"fileName":"SocksSocketImpl.java","nativeMethod":false,"methodName":"connect","className":"java.net.SocksSocketImpl","hCode":1780853535,"lineNumber":392,"class":"java.lang.StackTraceElement"},{"fileName":"Socket.java","nativeMethod":false,"methodName":"connect","className":"java.net.Socket","hCode":1700005482,"lineNumber":589,"class":"java.lang.StackTraceElement"},{"fileName":"PlainConnectionSocketFactory.java","nativeMethod":false,"methodName":"connectSocket","className":"org.apache.http.conn.socket.PlainConnectionSocketFactory","hCode":1175278798,"lineNumber":74,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultHttpClientConnectionOperator.java","nativeMethod":false,"methodName":"connect","className":"org.apache.http.impl.conn.DefaultHttpClientConnectionOperator","hCode":-502593941,"lineNumber":141,"class":"java.lang.StackTraceElement"},{"fileName":"PoolingHttpClientConnectionManager.java","nativeMethod":false,"methodName":"connect","className":"org.apache.http.impl.conn.PoolingHttpClientConnectionManager","hCode":-1370960713,"lineNumber":353,"class":"java.lang.StackTraceElement"},{"fileName":"MainClientExec.java","nativeMethod":false,"methodName":"establishRoute","className":"org.apache.http.impl.execchain.MainClientExec","hCode":848422720,"lineNumber":380,"class":"java.lang.StackTraceElement"},{"fileName":"MainClientExec.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.execchain.MainClientExec","hCode":1688513621,"lineNumber":236,"class":"java.lang.StackTraceElement"},{"fileName":"ProtocolExec.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.execchain.ProtocolExec","hCode":-642149983,"lineNumber":184,"class":"java.lang.StackTraceElement"},{"fileName":"RetryExec.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.execchain.RetryExec","hCode":2007799545,"lineNumber":88,"class":"java.lang.StackTraceElement"},{"fileName":"RedirectExec.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.execchain.RedirectExec","hCode":608278487,"lineNumber":110,"class":"java.lang.StackTraceElement"},{"fileName":"InternalHttpClient.java","nativeMethod":false,"methodName":"doExecute","className":"org.apache.http.impl.client.InternalHttpClient","hCode":2023045549,"lineNumber":184,"class":"java.lang.StackTraceElement"},{"fileName":"CloseableHttpClient.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.client.CloseableHttpClient","hCode":1628029009,"lineNumber":71,"class":"java.lang.StackTraceElement"},{"fileName":"CloseableHttpClient.java","nativeMethod":false,"methodName":"execute","className":"org.apache.http.impl.client.CloseableHttpClient","hCode":1628028993,"lineNumber":55,"class":"java.lang.StackTraceElement"},{"fileName":"ApacheHttpClient.java","nativeMethod":false,"methodName":"fallBackExecute","className":"org.openqa.selenium.remote.internal.ApacheHttpClient","hCode":652052258,"lineNumber":142,"class":"java.lang.StackTraceElement"},{"fileName":"ApacheHttpClient.java","nativeMethod":false,"methodName":"execute","className":"org.openqa.selenium.remote.internal.ApacheHttpClient","hCode":-904183858,"lineNumber":88,"class":"java.lang.StackTraceElement"},{"fileName":"ProtocolHandshake.java","nativeMethod":false,"methodName":"createSession","className":"org.openqa.selenium.remote.ProtocolHandshake","hCode":-960127848,"lineNumber":108,"class":"java.lang.StackTraceElement"},{"fileName":"ProtocolHandshake.java","nativeMethod":false,"methodName":"createSession","className":"org.openqa.selenium.remote.ProtocolHandshake","hCode":-960127892,"lineNumber":64,"class":"java.lang.StackTraceElement"},{"fileName":"HttpCommandExecutor.java","nativeMethod":false,"methodName":"execute","className":"org.openqa.selenium.remote.HttpCommandExecutor","hCode":-1852790188,"lineNumber":141,"class":"java.lang.StackTraceElement"},{"fileName":"DriverCommandExecutor.java","nativeMethod":false,"methodName":"execute","className":"org.openqa.selenium.remote.service.DriverCommandExecutor","hCode":-633333582,"lineNumber":82,"class":"java.lang.StackTraceElement"},{"fileName":"RemoteWebDriver.java","nativeMethod":false,"methodName":"execute","className":"org.openqa.selenium.remote.RemoteWebDriver","hCode":1767885088,"lineNumber":601,"class":"java.lang.StackTraceElement"},{"fileName":"RemoteWebDriver.java","nativeMethod":false,"methodName":"startSession","className":"org.openqa.selenium.remote.RemoteWebDriver","hCode":-1182648233,"lineNumber":241,"class":"java.lang.StackTraceElement"},{"fileName":"RemoteWebDriver.java","nativeMethod":false,"methodName":"<init>","className":"org.openqa.selenium.remote.RemoteWebDriver","hCode":2001602308,"lineNumber":128,"class":"java.lang.StackTraceElement"},{"fileName":"SafariDriver.java","nativeMethod":false,"methodName":"<init>","className":"org.openqa.selenium.safari.SafariDriver","hCode":-1792744904,"lineNumber":62,"class":"java.lang.StackTraceElement"},{"fileName":"SafariDriver.java","nativeMethod":false,"methodName":"<init>","className":"org.openqa.selenium.safari.SafariDriver","hCode":-1792744913,"lineNumber":53,"class":"java.lang.StackTraceElement"},{"fileName":"NativeConstructorAccessorImpl.java","nativeMethod":true,"methodName":"newInstance0","className":"sun.reflect.NativeConstructorAccessorImpl","hCode":-917798116,"lineNumber":-2,"class":"java.lang.StackTraceElement"},{"fileName":"NativeConstructorAccessorImpl.java","nativeMethod":false,"methodName":"newInstance","className":"sun.reflect.NativeConstructorAccessorImpl","hCode":-1864225098,"lineNumber":62,"class":"java.lang.StackTraceElement"},{"fileName":"DelegatingConstructorAccessorImpl.java","nativeMethod":false,"methodName":"newInstance","className":"sun.reflect.DelegatingConstructorAccessorImpl","hCode":-2122307259,"lineNumber":45,"class":"java.lang.StackTraceElement"},{"fileName":"Constructor.java","nativeMethod":false,"methodName":"newInstance","className":"java.lang.reflect.Constructor","hCode":-1319859919,"lineNumber":423,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultDriverProvider.java","nativeMethod":false,"methodName":"callConstructor","className":"org.openqa.selenium.remote.server.DefaultDriverProvider","hCode":-2096353668,"lineNumber":103,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultDriverProvider.java","nativeMethod":false,"methodName":"newInstance","className":"org.openqa.selenium.remote.server.DefaultDriverProvider","hCode":1201680335,"lineNumber":97,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultDriverFactory.java","nativeMethod":false,"methodName":"newInstance","className":"org.openqa.selenium.remote.server.DefaultDriverFactory","hCode":-1365021416,"lineNumber":60,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultSession.java","nativeMethod":false,"methodName":"call","className":"org.openqa.selenium.remote.server.DefaultSession$BrowserCreator","hCode":2056168471,"lineNumber":222,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultSession.java","nativeMethod":false,"methodName":"call","className":"org.openqa.selenium.remote.server.DefaultSession$BrowserCreator","hCode":2056168458,"lineNumber":209,"class":"java.lang.StackTraceElement"},{"fileName":"FutureTask.java","nativeMethod":false,"methodName":"run","className":"java.util.concurrent.FutureTask","hCode":-820313554,"lineNumber":266,"class":"java.lang.StackTraceElement"},{"fileName":"DefaultSession.java","nativeMethod":false,"methodName":"run","className":"org.openqa.selenium.remote.server.DefaultSession$1","hCode":-255143467,"lineNumber":176,"class":"java.lang.StackTraceElement"},{"fileName":"ThreadPoolExecutor.java","nativeMethod":false,"methodName":"runWorker","className":"java.util.concurrent.ThreadPoolExecutor","hCode":-1285150184,"lineNumber":1142,"class":"java.lang.StackTraceElement"},{"fileName":"ThreadPoolExecutor.java","nativeMethod":false,"methodName":"run","className":"java.util.concurrent.ThreadPoolExecutor$Worker","hCode":-242190645,"lineNumber":617,"class":"java.lang.StackTraceElement"},{"fileName":"Thread.java","nativeMethod":false,"methodName":"run","className":"java.lang.Thread","hCode":-82946320,"lineNumber":745,"class":"java.lang.StackTraceElement"}],"suppressed":[],"message":"Connection refused","hCode":882209463,"class":"java.net.ConnectException","screen":null},"class":"org.openqa.selenium.remote.Response","status":13}, Timed out
```

**Comment by azakus** [comment]
Only one version of safari is allowed to run under `safaridriver` at a time: https://webkit.org/blog/6900/webdriver-support-in-safari-10/, under the section `Safari Exclusive Safeguards`

**Comment by rictic** [comment]
Ok, I have something kinda hideous but working (gif at 3x speed because wct is slow), so this may not be a disaster:

![wct variants](https://cloud.githubusercontent.com/assets/1659/21246414/77f70332-c2dd-11e6-971d-98ccd3e91d6d.gif)


This changes the timings of some internals (BrowserRunner#browser isn't necessarily defined for a while), but idiomatic plugin writing shouldn't notice.

Now working on reducing from a pile of crazy hacks down to something reviewable.

**Comment by rictic** [comment]
Should be fixed by #460
